execute_tests.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+## Deploy virualenv for testing enironment molecule/ansible-playbook/infratest
+
+## Shell Opts ----------------------------------------------------------------
+
+set -x
+set -o pipefail
+export ANSIBLE_HOST_KEY_CHECKING=False
+
+## Variables -----------------------------------------------------------------
+
+SYS_VENV_NAME="${SYS_VENV_NAME:-venv-molecule}"
+SYS_CONSTRAINTS="constraints.txt"
+SYS_REQUIREMENTS="requirements.txt"
+SYS_INVENTORY="${SYS_INVENTORY:-/etc/openstack_deploy/openstack_inventory.json}"
+
+## Main ----------------------------------------------------------------------
+
+# Create virtualenv for molecule
+virtualenv --no-pip --no-setuptools --no-wheel "${SYS_VENV_NAME}"
+
+# Activate virtualenv
+source "${SYS_VENV_NAME}/bin/activate"
+
+# Ensure that correct pip version is installed
+PIP_TARGET="$(awk -F= '/^pip==/ {print $3}' ${SYS_CONSTRAINTS})"
+VENV_PYTHON="${SYS_VENV_NAME}/bin/python"
+VENV_PIP="${SYS_VENV_NAME}/bin/pip"
+
+if [[ "$(${VENV_PIP} --version)" != "pip ${PIP_TARGET}"* ]]; then
+    CURL_CMD="curl --silent --show-error --retry 5"
+    OUTPUT_FILE="get-pip.py"
+    ${CURL_CMD} https://bootstrap.pypa.io/get-pip.py > ${OUTPUT_FILE}  \
+        || ${CURL_CMD} https://raw.githubusercontent.com/pypa/get-pip/master/get-pip.py > ${OUTPUT_FILE}
+    GETPIP_OPTIONS="pip setuptools wheel --constraint ${SYS_CONSTRAINTS}"
```

Comment:
Are the constraints needed here? Since they are included in the requirements file, they will be applied on line 43.

Reply:
Yes, I think so, the constraints here are for making sure the correct version of **pip** is installed based on how the requirement file defines pip version(in this case it doesn't matter because our requirement file doesn't not have pip).
In line 43, ${PIP_OPTIONS} has `-r requirements.txt` and requirements.txt has `-c constraints.txt`, however, these constraints are a little bit different, they are telling pip to install the packages in requirements.txt such as molecule, pytest, flake 8 with correct versions defined in constraints.txt.